pandas/core/frame.py

```diff
@@ -1881,9 +1881,10 @@ def to_parquet(self, fname, engine='auto', compression='snappy',
             'pyarrow' is unavailable.
         compression : {'snappy', 'gzip', 'brotli', None}, default 'snappy'
             Name of the compression to use. Use ``None`` for no compression.
-        index : bool, default True
+        index : bool, default None
             If ``True``, include the dataframe's index(es) in the file output.
-            If ``False``, they will not be written to the file.
+            If ``False``, they will not be written to the file. If ``None``,
+            the behavior depends on the chosen engine.
```

Comment:
should we maybe shortly mention that if it is a non-default index, it will always be written with this default behaviour?

Reply:
I deliberately left that out since if we add support for another Parquet engine, or the implementation of one of the engines changes, our documentation would be out of date.